### DISCUSSION TITLE
How to create Variant Array

### DISCUSSION THREAD

**Discussion by simondarius** [FreeOpcUa/opcua-asyncio#1416] [Q&A] [answered]
Hi anyone knows if it is possible/how to create a variant array, an array that can store multiple datatypes like you see in the screenshot? When I do  " variant_array = ua.Variant(array_values, ua.VariantType.Variant "  I get " AttributeError: type object 'ObjectIds' has no attribute 'Variant' ". I've struggled with this and did not manage to find answers, any help would be apreciated!
![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/139557639/ce3024a9-4c35-492b-9099-fd774b6288fc)

**Comment by AndreasHeine** [comment, marked as answer]
dv = ua.DataValue(ua.Variant([200.0, 100.0], ua.VariantType.Double))

in opcua variables and properties containing a datavalue in the value attribute of a node which containes a variant. a variant is the value + the datatype (varianttype) + information about if its an array or not and if hints about size aswell


https://github.com/AndreasHeine/opcua-tutorial/blob/main/client/write.py
